tests/torch/test_knowledge_distillation.py

```diff
@@ -373,23 +371,48 @@ def forward(self, *args, **kwargs):
         return retval
 
 
-@pytest.mark.parametrize('shape_list', (
-    [(1, 2, 3, 4)],
-    [(1, 128)],
-    [(1, 128), (1, )]
-))
-def test_kd_softmax_loss_ignores_incompatible_outputs(shape_list: List[Tuple[int]]):
-    original_model = KDOutputModel(target_shapes=shape_list)
-    config = NNCFConfig.from_dict({
-        "input_info": {"sample_size": [1, 1, 1, 1]},
-        "compression": {
-            "algorithm": "knowledge_distillation",
-            "type": "softmax"
-        }
-    })
-    compressed_model = NNCFNetwork(original_model, [ModelInputInfo([1, 1, 1, 1])])
-    kd_builder = KnowledgeDistillationBuilder(config)
-    compressed_model = kd_builder.apply_to(compressed_model)
-    kd_ctrl = kd_builder.build_controller(compressed_model)
-    compressed_model.forward(torch.ones_like(compressed_model.mock_param))
-    kd_ctrl.loss()  # Should succeed - the loss for the incompatible outputs will be equal to 0
+class CustomOutputWeightenedModel(torch.nn.Module):
+    def __init__(self, input_shape: List[int], target_output_shapes: List[int]):
+        super().__init__()
+        self.weights = nn.ParameterDict()
+        self.linears = nn.ModuleDict()
+        self.target_output_shapes = target_output_shapes
+        self.linears['4'] = torch.nn.Linear(in_features=input_shape[3], out_features=input_shape[3])
+        self.linears['3'] = torch.nn.Linear(in_features=input_shape[3], out_features=1)
+        self.linears['2'] = torch.nn.Linear(in_features=input_shape[2], out_features=1)
+        self.linears['1'] = torch.nn.Linear(in_features=input_shape[1], out_features=1)
```

Comment:
Please refactor the test or add explicit comments explaining just what is happening here - it's hard to read the code handling in magic constants such as 4, 3, '4' etc

Reply:
Test has been changed